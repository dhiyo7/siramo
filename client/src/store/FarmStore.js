import { observable, computed } from 'mobx'
import { db, User } from './firebase'
import { Alert } from 'react-native'


class FarmStore {
  @observable farmData = {
    historyHumidity: [],
    historyWaterRatio: [],
    historyTemperature: [],
    historyWaterLevel: [],
    historyLoading: false
  }

  @observable navigation = {}

  getHistory = (key) => {
    this.farmData.historyLoading = true
    db.ref('/history/17UFak7JqufG1RXUeVW30jwdfrQ2').on('value', (snapshot) => {

      snapshot.forEach(snap => {
        this.farmData.historyHumidity.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().humidity)
        })
        this.farmData.historyWaterRatio.push({
          x: snap.val().last_updated,
          y: Math.round(parseFloat(snap.val().water_ratio))
        })
        this.farmData.historyTemperature.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().temperature)
        })
        this.farmData.historyWaterLevel.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().water_level)
        })
      })
      this.farmData.historyLoading = false
    })
  }

  setSchedule = (cronFormat, max, min) => {
    let updatedValue = {
      name: cronFormat,
      maxWaterRatio: max,
      minWaterRatio: min
    }
    console.log('Firebase Updated', updatedValue)
    db.ref(`/users`).update(updatedValue)
      .then(response => {
        console.log(response)
      })
      .catch(err => {
        console.log(err)
      })
  }

}

export default new FarmStore()