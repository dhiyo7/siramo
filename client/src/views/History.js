import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  View,
  FlatList,
  Dimensions
} from 'react-native';
import { inject, observer } from 'mobx-react'
import { TabViewAnimated, TabBar, SceneMap } from 'react-native-tab-view';
import Graph from '../components/farm/Graph'
import FarmStore from '../store/FarmStore'
import TabBarLogo from '../components/logo/TabBarLogo'
import Loader from '../components/customs/Loader'
import LandscapeView from 'react-native-landscape-view'
import { Ionicons } from '@expo/vector-icons'

const initialLayout = {
  height: 0,
  width: Dimensions.get('window').width,
};

const FirstRoute = () => <Graph sensorData={FarmStore.farmData.historyHumidity} humidityDetail={'humid'}/>
const SecondRoute = () => <Graph sensorData={FarmStore.farmData.historyTemperature} temperatureDetail={'panas'} color="blue"/>
const ThirdRoute = () => <Graph sensorData={FarmStore.farmData.historyWaterLevel} waterLevelDetail={'level'} color="red"/>
const FourthRoute = () => <Graph  sensorData={FarmStore.farmData.historyWaterRatio} waterRatioDetail={'water ratio'} color="green"/>

@inject('FarmStore')
@observer class History extends Component {

  constructor(props) {
    super(props);
    this.state = {  
      index: 0,
      routes: [
        { key: 'first', icon: 'md-cloudy'  },
        { key: 'second', icon: 'md-thermometer' },
        { key: 'third', icon: 'md-trending-up' },
        { key: 'fourth', icon: 'md-water' },
      ],
    };
  }

  static navigationOptions = {
    drawerLabel: 'History',
    drawerIcon: () => (
      <Ionicons name="md-analytics" size={24} color="green" />
    )
  }

  componentDidMount() {
    FarmStore.getHistory()
  }

  _handleIndexChange = index => this.setState({ index });

_renderHeader = props => <TabBar  {...props} renderIcon={this._renderIcon} tabStyle={{backgroundColor: '#AED581'}}/>;

  _renderScene = SceneMap({
    first: FirstRoute,
    second: SecondRoute,
    third: ThirdRoute,
    fourth: FourthRoute
  })

  _renderIcon = ({ route }: any) => {
    return <Ionicons name={route.icon} size={24}  />;
  };

  render() {
    const loading = FarmStore.farmData.historyLoading
    if (!loading) {
      return (
        <TabViewAnimated
          navigationState={this.state}
          renderScene={this._renderScene}
          renderHeader={this._renderHeader}
          onIndexChange={this._handleIndexChange}
          initialLayout={initialLayout}
        />
      );
    } else {
      return <Loader loading={loading}/>
    }
    
  }
}

export default History