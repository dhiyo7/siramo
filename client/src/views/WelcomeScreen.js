import React, { Component } from 'react';
import { View, StyleSheet, Image, AsyncStorage} from 'react-native'
import UserStore from '../store/UserStore'
import FarmStore from '../store/FarmStore'
import { User } from '../store/firebase'
import { observer } from 'mobx-react'

const background = require('../assets/logo/Drawing1-Model3.png')

@observer class WelcomeScreen extends Component {
  
  static navigationOptions = {
    header: null
  }
  componentDidMount = () => {
    setTimeout( async () => {
      let uid = await AsyncStorage.getItem('userId')
      let email = await AsyncStorage.getItem('email')
      await UserStore.assignUserData({
        uid,
        email
      })
      await FarmStore.getFarmData()
      uid == null?
      this.props.navigation.push('Login'):this.props.navigation.push('Home')
    }, 3000)
  }
  render() {
    return (
      <View style={styles.mainContainer}>
          <View style={styles.container}>
          <Image
            style={styles.imageContainer}
            source={background}
          />
          </View>
      </View>
    ); 
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: '#A1887F',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%'
  },
  container: {
    width: 350,
    alignItems: 'center',
    justifyContent: 'center',
  },
  imageContainer: {
    width: 310,
    height: 230,
  },
})


export default WelcomeScreen;