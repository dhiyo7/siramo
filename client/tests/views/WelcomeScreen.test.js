// import 'jsdom-global/register'
import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import MockStorage from '../MockStorage';
import 'jsdom-global/register'

const storageCache = {};
const AsyncStorage = new MockStorage(storageCache);
import { View, Image } from 'react-native'
import History from '../../src/views/History'
import FarmStore from '../../src/store/FarmStore'
import WelcomeScreen from '../../src/views/WelcomeScreen';

Enzyme.configure({ adapter: new Adapter()})

describe('render components', () => {
  it('has view and image', () => {
      const wrapper = shallow(
        <WelcomeScreen />
      )
      const component = renderer.create(<WelcomeScreen />)
      expect(wrapper.containsMatchingElement([
        <View/>,
        <Image/>,
      ]))
      expect(component).toMatchSnapshot()
  })

  test('component did mount when its rendering', () => {
    const navigation = { navigate: jest.fn() }
    const wrapper = mount(<WelcomeScreen navigation={navigation}/>)
    wrapper.instance().componentDidMount()
  })
})