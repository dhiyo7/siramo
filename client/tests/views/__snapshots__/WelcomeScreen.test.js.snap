// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render components has view and image 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#A1887F",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 350,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 230,
          "width": 310,
        }
      }
    />
  </View>
</View>
`;
